Partition-selection drop-downs don't work properly for non-String columns

<!-- Please read our Code of Conduct: https://github.com/deephaven/deephaven-core/blob/main/CODE_OF_CONDUCT.md -->
<!-- Please search existing issues to avoid creating duplicates. -->
<!-- Use clear and concise text to keep the issue brief and to the point. -->

**Description**

<!-- Description of what the bug is. -->

When using non-`String` partitioning columns (`long` in my example), the partition-selection drop-down incorrectly tries to filter as if the values are strings. It wraps them in quotes, causing `MatchFilter` parsing to reject the literal.

**Steps to reproduce**

<!--
A numbered set of steps to reproduce the issue, e.g.

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
-->

Use a command like:
```
t=io.deephaven.parquet.table.ParquetTools.readTable(path)
```
to load a Parquet table that uses metadata files, a `TableDefinition`, or inferred partitioning column types from key=value directory name pairs. If any of the partitioning column types is not a `String`, you should see this error or similar.

Note that the automatic first-partition retrieval fails, as does selecting a partition to filter to. `coalesce()` or `where()` on the table to skip this servers as a workaround.
 
**Expected results**

<!-- 
What you expected to happen. Index it by the steps to reproduce to make it clear, e.g.

3. New page to load
-->

The table to render properly.

**Actual results**

<!-- 
What actually happened. Index it by the steps to reproduce to make it clear, e.g.

3. New page did not load
-->

A table widget with an error loaded. See screeenshot.

**Additional details and attachments**

<!-- If applicable, add any additional screenshots, logs, or other attachments to help explain your problem. -->

Here's the exception stack trace from the server-side, as displayed in the console:
```
heduler-Concurrent-3 | i.d.s.s.SessionService    | Internal Error 'ef498c1c-f351-4b5a-9f2b-da5f85b0e713' java.lang.IllegalArgumentException: Failed to convert literal value <"100"> for column "cnum" of type long
	at io.deephaven.engine.table.impl.select.MatchFilter.init(MatchFilter.java:155)
	at io.deephaven.engine.table.impl.PartitionAwareSourceTable.whereImpl(PartitionAwareSourceTable.java:288)
	at io.deephaven.engine.table.impl.PartitionAwareSourceTable.where(PartitionAwareSourceTable.java:272)
	at io.deephaven.engine.table.impl.PartitionAwareSourceTable.where(PartitionAwareSourceTable.java:35)
	at io.deephaven.server.table.ops.FilterTableGrpcImpl.create(FilterTableGrpcImpl.java:57)
	at io.deephaven.server.table.ops.FilterTableGrpcImpl.create(FilterTableGrpcImpl.java:30)
	at io.deephaven.server.table.ops.TableServiceGrpcImpl$BatchExportBuilder.doExport(TableServiceGrpcImpl.java:689)
	at io.deephaven.server.table.ops.TableServiceGrpcImpl.lambda$batch$5(TableServiceGrpcImpl.java:541)
	at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:921)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.deephaven.server.runner.scheduler.SchedulerModule$ThreadFactory.lambda$newThread$0(SchedulerModule.java:78)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NumberFormatException: For input string: ""100""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Long.parseLong(Long.java:678)
	at java.base/java.lang.Long.parseLong(Long.java:817)
	at io.deephaven.engine.table.impl.select.MatchFilter$ColumnTypeConvertorFactory$4.convertStringLiteral(MatchFilter.java:229)
	at io.deephaven.engine.table.impl.select.MatchFilter.init(MatchFilter.java:152)
	... 15 more
```

Here's a screenshot of the issue:
![Screenshot 2023-08-08 at 9 43 32 AM](https://github.com/deephaven/web-client-ui/assets/5342661/2a3504d4-f10f-4a14-8ca5-4f593d3065c1)

**Versions**

<!-- Version information. These versions can be copied from the Settings panel in the Deephaven UI. -->
Version
0.27.0

--- a/packages/iris-grid/src/IrisGrid.tsx
+++ b/packages/iris-grid/src/IrisGrid.tsx
@@ -1921,38 +1921,27 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
   }
 
   updatePartition(partition: string, partitionColumn: Column): void {
-    if (partitionColumn.type === 'char' && partition === '') {
+    if (TableUtils.isCharType(partitionColumn.type) && partition === '') {
       return;
     }
 
     const { model } = this.props;
 
-    let partitionFilterValue;
-    if (
-      partitionColumn.type === 'java.lang.String' ||
-      partitionColumn.type === 'char'
-    ) {
-      partitionFilterValue = model.dh.FilterValue.ofString(
-        model.displayString(
+    const partitionText = TableUtils.isCharType(partitionColumn.type)
+      ? model.displayString(
           partition,
           partitionColumn.type,
           partitionColumn.name
         )
-      );
-    } else if (
-      partitionColumn.type === 'java.lang.Boolean' ||
-      partitionColumn.type === 'boolean'
-    ) {
-      partitionFilterValue = model.dh.FilterValue.ofBoolean(
-        typeof partition === 'string'
-          ? TableUtils.makeBooleanValue(partition)
-          : partition
-      );
-    } else {
-      partitionFilterValue = model.dh.FilterValue.ofNumber(partition);
-    }
+      : partition;
+    const partitionFilter = this.tableUtils.makeQuickFilterFromComponent(
+      partitionColumn,
+      partitionText
+    );
 
-    const partitionFilter = partitionColumn.filter().eq(partitionFilterValue);
+    if (partitionFilter === null) {
+      return;
+    }
 
     const partitionFilters = [partitionFilter];
     this.setState({

--- a/packages/iris-grid/src/IrisGrid.tsx
+++ b/packages/iris-grid/src/IrisGrid.tsx
@@ -1938,7 +1938,6 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
       partitionColumn,
       partitionText
     );
-
     if (partitionFilter === null) {
       return;
     }

--- a/packages/iris-grid/src/IrisGrid.tsx
+++ b/packages/iris-grid/src/IrisGrid.tsx
@@ -1921,10 +1921,25 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
   }
 
   updatePartition(partition: string, partitionColumn: Column): void {
+    if (partitionColumn.type === 'char' && partition === '') {
+      return;
+    }
+
     const { model } = this.props;
     const partitionFilter = partitionColumn
       .filter()
-      .eq(model.dh.FilterValue.ofString(partition));
+      .eq(
+        partitionColumn.type === 'java.lang.String' ||
+          partitionColumn.type === 'char'
+          ? model.dh.FilterValue.ofString(
+              model.displayString(
+                partition,
+                partitionColumn.type,
+                partitionColumn.name
+              )
+            )
+          : model.dh.FilterValue.ofNumber(partition)
+      );
     const partitionFilters = [partitionFilter];
     this.setState({
       partition,
@@ -4424,7 +4439,7 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
                     type: string,
                     stringName: string
                   ) => model.displayString(value, type, stringName)}
-                  columnName={partitionColumn.name}
+                  column={partitionColumn}
                   partition={partition}
                   onChange={this.handlePartitionChange}
                   onFetchAll={this.handlePartitionFetchAll}

--- a/packages/iris-grid/src/IrisGrid.tsx
+++ b/packages/iris-grid/src/IrisGrid.tsx
@@ -1943,7 +1943,11 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
       partitionColumn.type === 'java.lang.Boolean' ||
       partitionColumn.type === 'boolean'
     ) {
-      partitionFilterValue = model.dh.FilterValue.ofBoolean(partition);
+      partitionFilterValue = model.dh.FilterValue.ofBoolean(
+        typeof partition === 'string'
+          ? TableUtils.makeBooleanValue(partition)
+          : partition
+      );
     } else {
       partitionFilterValue = model.dh.FilterValue.ofNumber(partition);
     }

--- a/packages/iris-grid/src/IrisGrid.tsx
+++ b/packages/iris-grid/src/IrisGrid.tsx
@@ -1926,20 +1926,30 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
     }
 
     const { model } = this.props;
-    const partitionFilter = partitionColumn
-      .filter()
-      .eq(
-        partitionColumn.type === 'java.lang.String' ||
-          partitionColumn.type === 'char'
-          ? model.dh.FilterValue.ofString(
-              model.displayString(
-                partition,
-                partitionColumn.type,
-                partitionColumn.name
-              )
-            )
-          : model.dh.FilterValue.ofNumber(partition)
+
+    let partitionFilterValue;
+    if (
+      partitionColumn.type === 'java.lang.String' ||
+      partitionColumn.type === 'char'
+    ) {
+      partitionFilterValue = model.dh.FilterValue.ofString(
+        model.displayString(
+          partition,
+          partitionColumn.type,
+          partitionColumn.name
+        )
       );
+    } else if (
+      partitionColumn.type === 'java.lang.Boolean' ||
+      partitionColumn.type === 'boolean'
+    ) {
+      partitionFilterValue = model.dh.FilterValue.ofBoolean(partition);
+    } else {
+      partitionFilterValue = model.dh.FilterValue.ofNumber(partition);
+    }
+
+    const partitionFilter = partitionColumn.filter().eq(partitionFilterValue);
+
     const partitionFilters = [partitionFilter];
     this.setState({
       partition,

--- a/packages/iris-grid/src/IrisGridPartitionSelector.test.tsx
+++ b/packages/iris-grid/src/IrisGridPartitionSelector.test.tsx
@@ -14,7 +14,7 @@ function makeIrisGridPartitionSelector(
     <IrisGridPartitionSelector
       dh={dh}
       table={table}
-      columnName="0"
+      column={new IrisGridTestUtils(dh).makeColumn()}
       getFormattedString={getFormattedString}
       onChange={onChange}
       onDone={onDone}

--- a/packages/iris-grid/src/IrisGridPartitionSelector.tsx
+++ b/packages/iris-grid/src/IrisGridPartitionSelector.tsx
@@ -4,10 +4,9 @@ import { DropdownMenu, Tooltip } from '@deephaven/components';
 import { vsTriangleDown, vsClose } from '@deephaven/icons';
 import Log from '@deephaven/log';
 import debounce from 'lodash.debounce';
-import type { dh as DhType, Table } from '@deephaven/jsapi-types';
+import type { Column, dh as DhType, Table } from '@deephaven/jsapi-types';
 import PartitionSelectorSearch from './PartitionSelectorSearch';
 import './IrisGridPartitionSelector.scss';
-import { ColumnName } from './CommonTypes';
 
 const log = Log.module('IrisGridPartitionSelector');
 
@@ -16,7 +15,7 @@ interface IrisGridPartitionSelectorProps<T> {
   dh: DhType;
   getFormattedString: (value: T, type: string, name: string) => string;
   table: Table;
-  columnName: ColumnName;
+  column: Column;
   partition: string;
   onAppend?: (partition: string) => void;
   onFetchAll: () => void;
@@ -90,9 +89,22 @@ class IrisGridPartitionSelector<T> extends Component<
   handlePartitionChange(event: React.ChangeEvent<HTMLInputElement>): void {
     log.debug2('handlePartitionChange');
 
+    const { column } = this.props;
     const { value: partition } = event.target;
 
-    this.setState({ partition });
+    this.setState({
+      partition:
+        column.type === 'char' && partition.length > 0
+          ? partition.charCodeAt(0).toString()
+          : partition,
+
+      // partition:
+      //   column.type === 'char'
+      //     ? partition.length > 0
+      //       ? partition.charCodeAt(0).toString()
+      //       : prevState.partition
+      //     : partition,
+    });
 
     this.debounceUpdate();
   }
@@ -154,7 +166,7 @@ class IrisGridPartitionSelector<T> extends Component<
   }
 
   render(): JSX.Element {
-    const { columnName, dh, getFormattedString, onAppend, onDone, table } =
+    const { column, dh, getFormattedString, onAppend, onDone, table } =
       this.props;
     const { partition } = this.state;
     const partitionSelectorSearch = (
@@ -173,12 +185,16 @@ class IrisGridPartitionSelector<T> extends Component<
     return (
       <div className="iris-grid-partition-selector">
         <div className="status-message">
-          <span>Filtering &quot;{columnName}&quot; partition to</span>
+          <span>Filtering &quot;{column.name}&quot; partition to</span>
         </div>
         <div className="input-group">
           <input
             type="text"
-            value={partition}
+            value={
+              column.type === 'char' && partition.toString().length > 0
+                ? String.fromCharCode(parseInt(partition, 10))
+                : partition
+            }
             onChange={this.handlePartitionChange}
             className="form-control input-partition"
           />

--- a/packages/iris-grid/src/IrisGridPartitionSelector.tsx
+++ b/packages/iris-grid/src/IrisGridPartitionSelector.tsx
@@ -97,13 +97,6 @@ class IrisGridPartitionSelector<T> extends Component<
         column.type === 'char' && partition.length > 0
           ? partition.charCodeAt(0).toString()
           : partition,
-
-      // partition:
-      //   column.type === 'char'
-      //     ? partition.length > 0
-      //       ? partition.charCodeAt(0).toString()
-      //       : prevState.partition
-      //     : partition,
     });
 
     this.debounceUpdate();

--- a/packages/iris-grid/src/IrisGridPartitionSelector.tsx
+++ b/packages/iris-grid/src/IrisGridPartitionSelector.tsx
@@ -185,7 +185,6 @@ class IrisGridPartitionSelector<T> extends Component<
           <input
             type="text"
             value={
-              // IrisGridUtils.convertValueToText(partition, column.type)
               column.type === 'char' && partition.toString().length > 0
                 ? String.fromCharCode(parseInt(partition, 10))
                 : IrisGridUtils.convertValueToText(partition, column.type)

--- a/packages/iris-grid/src/IrisGridPartitionSelector.tsx
+++ b/packages/iris-grid/src/IrisGridPartitionSelector.tsx
@@ -7,6 +7,7 @@ import debounce from 'lodash.debounce';
 import type { Column, dh as DhType, Table } from '@deephaven/jsapi-types';
 import PartitionSelectorSearch from './PartitionSelectorSearch';
 import './IrisGridPartitionSelector.scss';
+import IrisGridUtils from './IrisGridUtils';
 
 const log = Log.module('IrisGridPartitionSelector');
 
@@ -184,9 +185,10 @@ class IrisGridPartitionSelector<T> extends Component<
           <input
             type="text"
             value={
+              // IrisGridUtils.convertValueToText(partition, column.type)
               column.type === 'char' && partition.toString().length > 0
                 ? String.fromCharCode(parseInt(partition, 10))
-                : partition
+                : IrisGridUtils.convertValueToText(partition, column.type)
             }
             onChange={this.handlePartitionChange}
             className="form-control input-partition"

--- a/packages/iris-grid/src/IrisGridPartitionSelector.tsx
+++ b/packages/iris-grid/src/IrisGridPartitionSelector.tsx
@@ -5,6 +5,7 @@ import { vsTriangleDown, vsClose } from '@deephaven/icons';
 import Log from '@deephaven/log';
 import debounce from 'lodash.debounce';
 import type { Column, dh as DhType, Table } from '@deephaven/jsapi-types';
+import { TableUtils } from '@deephaven/jsapi-utils';
 import PartitionSelectorSearch from './PartitionSelectorSearch';
 import './IrisGridPartitionSelector.scss';
 import IrisGridUtils from './IrisGridUtils';
@@ -95,7 +96,7 @@ class IrisGridPartitionSelector<T> extends Component<
 
     this.setState({
       partition:
-        column.type === 'char' && partition.length > 0
+        TableUtils.isCharType(column.type) && partition.length > 0
           ? partition.charCodeAt(0).toString()
           : partition,
     });
@@ -185,7 +186,8 @@ class IrisGridPartitionSelector<T> extends Component<
           <input
             type="text"
             value={
-              column.type === 'char' && partition.toString().length > 0
+              TableUtils.isCharType(column.type) &&
+              partition.toString().length > 0
                 ? String.fromCharCode(parseInt(partition, 10))
                 : IrisGridUtils.convertValueToText(partition, column.type)
             }

--- a/packages/iris-grid/src/PartitionSelectorSearch.tsx
+++ b/packages/iris-grid/src/PartitionSelectorSearch.tsx
@@ -233,9 +233,14 @@ class PartitionSelectorSearch<T> extends Component<
   handleTextChange(event: React.ChangeEvent<HTMLInputElement>): void {
     log.debug2('handleTextChange');
 
+    const { table } = this.props;
     const { value: text } = event.target;
 
-    this.setState({ text });
+    if (text !== '' && TableUtils.isIntegerType(table.columns[0].type)) {
+      this.setState({ text: parseInt(text, 10).toString() });
+    } else {
+      this.setState({ text });
+    }
 
     this.debounceUpdateFilter();
   }
@@ -287,23 +292,16 @@ class PartitionSelectorSearch<T> extends Component<
     const filters = [];
     if (filterText.length > 0) {
       const column = table.columns[0];
-      if (
-        column.type === 'java.lang.String' ||
-        column.type === 'char' ||
-        column.type === 'java.lang.Boolean' ||
-        !Number.isNaN(Number(filterText))
-      ) {
-        const filter = this.tableUtils.makeQuickFilterFromComponent(
-          column,
-          column.type === 'java.lang.String' ? `~${filterText}` : filterText
+      const filter = this.tableUtils.makeQuickFilterFromComponent(
+        column,
+        TableUtils.isStringType(column.type) ? `~${filterText}` : filterText
+      );
+      if (!filter) {
+        throw new Error(
+          'Unable to create column filter for partition selector'
         );
-        if (!filter) {
-          throw new Error(
-            'Unable to create column filter for partition selector'
-          );
-        }
-        filters.push(filter);
       }
+      filters.push(filter);
     }
 
     log.debug2('updateFilter', filters);
@@ -313,17 +311,22 @@ class PartitionSelectorSearch<T> extends Component<
   }
 
   render(): JSX.Element {
+    const { table } = this.props;
     const { isLoading, itemCount, items, offset, text } = this.state;
+
     const listHeight =
       Math.min(itemCount, PartitionSelectorSearch.MAX_VISIBLE_ITEMS) *
         ItemList.DEFAULT_ROW_HEIGHT +
       // Adjust for ListItem vertical padding - .375rem ~ 5.25px
       11;
+    const inputType = TableUtils.isNumberType(table.columns[0].type)
+      ? 'number'
+      : 'text';
     return (
       <div className="iris-grid-partition-selector-search">
         <div className="search-container">
           <input
-            type="text"
+            type={inputType}
             ref={searchInput => {
               this.searchInput = searchInput;
             }}

--- a/packages/iris-grid/src/PartitionSelectorSearch.tsx
+++ b/packages/iris-grid/src/PartitionSelectorSearch.tsx
@@ -75,7 +75,6 @@ class PartitionSelectorSearch<T> extends Component<
     super(props);
 
     this.handleFilterChange = this.handleFilterChange.bind(this);
-    this.handleInputFocus = this.handleInputFocus.bind(this);
     this.handleKeyDown = this.handleKeyDown.bind(this);
     this.handleListKeydown = this.handleListKeydown.bind(this);
     this.handleSelect = this.handleSelect.bind(this);
@@ -185,12 +184,6 @@ class PartitionSelectorSearch<T> extends Component<
     const { table } = this.props;
     const itemCount = table.size;
     this.setState({ itemCount, isLoading: true });
-  }
-
-  handleInputFocus(): void {
-    if (this.itemList) {
-      this.itemList.focusItem(0);
-    }
   }
 
   handleSelect(itemIndex: ModelIndex): void {
@@ -333,7 +326,6 @@ class PartitionSelectorSearch<T> extends Component<
             value={text}
             placeholder="Available Partitions"
             onChange={this.handleTextChange}
-            onFocus={this.handleInputFocus}
             onKeyDown={this.handleKeyDown}
             className="form-control input-partition"
           />

--- a/packages/iris-grid/src/PartitionSelectorSearch.tsx
+++ b/packages/iris-grid/src/PartitionSelectorSearch.tsx
@@ -287,6 +287,13 @@ class PartitionSelectorSearch<T> extends Component<
     const filters = [];
     if (filterText.length > 0) {
       const column = table.columns[0];
+      if (
+        column.type !== 'java.lang.String' &&
+        column.type !== 'char' &&
+        Number.isNaN(Number(filterText))
+      ) {
+        return;
+      }
       const filter = this.tableUtils.makeQuickFilterFromComponent(
         column,
         column.type === 'java.lang.String' ? `~${filterText}` : filterText

--- a/packages/iris-grid/src/PartitionSelectorSearch.tsx
+++ b/packages/iris-grid/src/PartitionSelectorSearch.tsx
@@ -297,9 +297,12 @@ class PartitionSelectorSearch<T> extends Component<
           column,
           column.type === 'java.lang.String' ? `~${filterText}` : filterText
         );
-        if (filter) {
-          filters.push(filter);
+        if (!filter) {
+          throw new Error(
+            'Unable to create column filter for partition selector'
+          );
         }
+        filters.push(filter);
       }
     }
 

--- a/packages/iris-grid/src/PartitionSelectorSearch.tsx
+++ b/packages/iris-grid/src/PartitionSelectorSearch.tsx
@@ -1,6 +1,7 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import debounce from 'lodash.debounce';
+import { TableUtils } from '@deephaven/jsapi-utils';
 import type { dh as DhType, Table } from '@deephaven/jsapi-types';
 import { ItemList, LoadingSpinner } from '@deephaven/components';
 import Log from '@deephaven/log';
@@ -86,6 +87,9 @@ class PartitionSelectorSearch<T> extends Component<
     this.searchInput = null;
     this.timer = null;
 
+    const { dh } = props;
+    this.tableUtils = new TableUtils(dh);
+
     this.state = {
       offset: 0,
       itemCount: 0,
@@ -124,6 +128,8 @@ class PartitionSelectorSearch<T> extends Component<
   searchInput: HTMLInputElement | null;
 
   timer: null;
+
+  tableUtils: TableUtils;
 
   handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>): boolean {
     if (this.itemList == null) {
@@ -275,19 +281,19 @@ class PartitionSelectorSearch<T> extends Component<
   }
 
   updateFilter(): void {
-    const { dh, initialPageSize, table } = this.props;
+    const { initialPageSize, table } = this.props;
     const { text } = this.state;
     const filterText = text.trim();
     const filters = [];
     if (filterText.length > 0) {
       const column = table.columns[0];
-      const filter = column
-        .filter()
-        .invoke(
-          'matches',
-          dh.FilterValue.ofString(`(?s)(?i).*\\Q${filterText}\\E.*`)
-        );
-      filters.push(filter);
+      const filter = this.tableUtils.makeQuickFilterFromComponent(
+        column,
+        column.type === 'java.lang.String' ? `~${filterText}` : filterText
+      );
+      if (filter !== null) {
+        filters.push(filter);
+      }
     }
 
     log.debug2('updateFilter', filters);

--- a/packages/iris-grid/src/PartitionSelectorSearch.tsx
+++ b/packages/iris-grid/src/PartitionSelectorSearch.tsx
@@ -288,18 +288,18 @@ class PartitionSelectorSearch<T> extends Component<
     if (filterText.length > 0) {
       const column = table.columns[0];
       if (
-        column.type !== 'java.lang.String' &&
-        column.type !== 'char' &&
-        Number.isNaN(Number(filterText))
+        column.type === 'java.lang.String' ||
+        column.type === 'char' ||
+        column.type === 'java.lang.Boolean' ||
+        !Number.isNaN(Number(filterText))
       ) {
-        return;
-      }
-      const filter = this.tableUtils.makeQuickFilterFromComponent(
-        column,
-        column.type === 'java.lang.String' ? `~${filterText}` : filterText
-      );
-      if (filter !== null) {
-        filters.push(filter);
+        const filter = this.tableUtils.makeQuickFilterFromComponent(
+          column,
+          column.type === 'java.lang.String' ? `~${filterText}` : filterText
+        );
+        if (filter) {
+          filters.push(filter);
+        }
       }
     }
 

